plugins/optimization-detective/class-od-preload-link-collection.php

```diff
@@ -166,6 +172,31 @@ public function get_html(): string {
 		return implode( '', $link_tags );
 	}
 
+	/**
+	 * Constructs the Link HTTP response header.
+	 *
+	 * @return string|null Link HTTP response header, or null if there are none.
+	 */
+	public function get_response_header(): ?string {
+		$link_headers = array();
+
+		foreach ( $this->prepare_links() as $link ) {
+			$link_header = '<' . esc_url_raw( $link['attributes']['href'] ?? '' ) . '>; rel="preload"';
```

Comment:
Is it valid to send a `Link` without the header? It seems that the [uri-reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link#uri-reference) is required. In reality, the only reason why `href` was made optional was for the sake of preloading responsive images since Safari at the time didn't support `imagesrcset` and `imagesizes`, so adding a preload link that omitted the `href` was a way to prevent Safari from needlessly preloading an image which might not be used. 

Nevertheless, as seen in [this web.dev article](https://web.dev/articles/preload-responsive-images#image-set:~:text=Responsive%20image%20preloading%20lets%20you%20load%20those%20images%20faster.), there is still a use case for a preload link without `href` and that is for preloading background images which have different pixel densities:

```html
<link rel="preload" as="image" imagesrcset="cat.png 1x, cat-2x.png 2x">
```

So it seems we do need `href` to remain able to be optional.

What to do in the case where there is no `href`?  I just made a [test glitch](https://imagesrcset-only-preload-link-http-header.glitch.me/) to check various scenarios. You can see that [when an empty string is used](https://imagesrcset-only-preload-link-http-header.glitch.me/?preload_url_type=empty-string), there is nothing downloaded:

![image](https://github.com/WordPress/performance/assets/134745/6174a595-b160-45a2-a497-0f3b7741e536)

In contrast, check what happens [when `about:blank` is used as the URL](https://imagesrcset-only-preload-link-http-header.glitch.me/?preload_url_type=about-blank):

![image](https://github.com/WordPress/performance/assets/134745/581b8b08-a0f5-40f4-8c91-77f5ae3f88f2)

It works!

So let's do this:

```suggestion
			// The about:blank is present since a Link without a reference-uri is invalid so any imagesrcset would otherwise not get downloaded.
			$link_header = '<' . esc_url_raw( $link['attributes']['href'] ?? 'about:blank' ) . '>; rel="preload"';
```

Reply:
I also checked and using `about:blank` here works in Firefox and Safari too. This doesn't seem to be part of the spec, so it could be an implementation accident. But the alternative is to skip sending the `Link` altogether. So I think using `about:blank` is best.

Reply:
I see, thanks this was very insightful.